Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfileController.cs

```diff
@@ -162,5 +162,46 @@ public async Task<ActionResult> UpdateAsync([FromBody] ProfileApiModel apiModel)
                 return BadRequest();
             }
         }
+
+        [HttpPost("follow/{user=user}")]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> Follow(ProfileApiModel user)
+        {
+            // Get emails of each user
+            string thisUserEmail = GetUserEmail();
+            string followEmail = user.Email;
+            if (thisUserEmail is null)
+            {
+                return NotFound(thisUserEmail);
+            }
+            if (followEmail is null)
+            {
+                return NotFound(followEmail);
+            }
+            // Get users into domain models
+            DomainProfile thisUser;
+            DomainProfile followUser;
+            try
+            {
+                var usersQuery = await _repository.GetProfilesByEmailAsync(new List<string> { thisUserEmail, followEmail });
+                thisUser = usersQuery.Single(x => x.Email == thisUserEmail);
+                followUser = usersQuery.Single(x => x.Email == followEmail);
+                // Add following relationships
+                thisUser.FollowingEmails.Add(followEmail);
+                followUser.FollowerEmails.Add(thisUserEmail);
+                // Update in the database
+                await _repository.UpdateProfileAsync(thisUserEmail, thisUser);
+                await _repository.UpdateProfileAsync(followEmail, followUser);
+                return Ok();
+            }
+            catch (ArgumentException e)
+            {
+                _logger.LogError(e.Message);
+                _logger.LogError(e.StackTrace);
```

Comment:
all log methods have overloads for passing a whole exception object, optionally including a contextual message. in general just use those. this is good enough for now though

Reply:
Unchanged